### ISSUE TITLE
MacOS 10.15 Requirement?

### ISSUE THREAD

**Issue by bdkjones** [vapor/websocket-kit#80]
Websocket-Kit requires a deployment target of macOS 10.15+

I read that Vapor 4 requires 10.15+ because it replaced OpenSSL with the Apple's new CryptoKit library, but I was hoping to use Websocket-Kit as a stand-alone package (outside of Vapor) because I cannot yet limit my app's deployment target to 10.15+.

Is that possible? Looking at the source, I see some TLSConfiguration in `websocketClient` and the extension that adds the various `connect()` functions, but it looks like the **server** side of things is in the clear for older macOS versions? Is that a safe conclusion?

(Apologies for asking a question as an issue; it's just VERY hard to reach anyone with appropriate expertise any other way. I DID search!)

**Comment by 0xTim** [comment]
Hey @bdkjones! For all the Vapor 4 packages we made the decision to set the minimum macOS version for all packages so that if we needed to pull in Crypto (or any future package) we wouldn't have to wait until a major version bump (I think this was discussed on the forums). However, you are right in that there's nothing actually requiring 10.15 in Websockets currently. The easiest way would be to fork this package and just remove the dependency, or use NIO's websockets directly
